terraform/charm/large_deployment/main.tf

```diff
@@ -0,0 +1,185 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+locals {
+  apps = [
+    for app in concat(var.apps != null ? var.apps : []) : app if app != null
+  ]
+
+  apps_in_main_model = [
+    for app in concat([var.failover], local.apps) :
+    app if app != null && app.model == var.main.model
+  ]
+  apps_not_in_main_model = [
+    for app in concat([var.failover], local.apps) :
+    app if app != null && app.model != var.main.model
+  ]
+  apps_in_failover_model = [
+    for app in local.apps :
+    app if app.model == var.failover.model
+  ]
+  apps_not_in_failover_model = [
+    for app in local.apps :
+    app if app.model != var.failover.model
+  ]
+
+  all_models = distinct(concat(
+    [var.main.model],
+    var.failover != null ? [var.failover.model] : [],
+    var.apps != null ? [for app in var.apps : app.model] : [],
+  ))
+}
+
+#--------------------------------------------------------
+# 1. DEPLOYMENTS
+#--------------------------------------------------------
+
+# main orchestrator opensearch app
+module "opensearch_main" {
+  source = "../simple_deployment"
+
+  channel  = var.main.channel
+  revision = var.main.revision
+  base     = var.main.base
+
+  app_name          = var.main.app_name
+  units             = var.main.units
+  config            = merge(var.main.config, { "cluster_name" : var.cluster_name, "init_hold" : "false" })
```

Comment:
Should we let users configure init-hold?

Reply:
No, this is a config option that if not used correctly, will prevent the creation of the cluster - so we manage it ourselves